vaadin-messages-flow-parent/vaadin-messages-flow/src/main/java/com/vaadin/flow/component/messages/MessageList.java

```diff
@@ -12,27 +12,114 @@
  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  * License for the specific language governing permissions and limitations under
  * the License.
- *
  */
 package com.vaadin.flow.component.messages;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 import com.vaadin.flow.component.Component;
 import com.vaadin.flow.component.HasSize;
 import com.vaadin.flow.component.HasStyle;
 import com.vaadin.flow.component.Tag;
-import com.vaadin.flow.component.Text;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.internal.JsonUtils;
+
+import elemental.json.JsonArray;
 
 /**
- * Server-side component for the {@code vaadin-message-list} element.
+ * Server-side component for the {@code vaadin-message-list} element. The
+ * component displays a list of messages that can be configured with
+ * {@link #setItems(Collection)}.
  *
  * @author Vaadin Ltd.
  */
 @Tag("vaadin-message-list")
-// @JsModule("@vaadin/vaadin-messages/src/vaadin-message-list.js")
-// @NpmPackage(value = "@vaadin/vaadin-messages", version = "1.0.0")
+@JsModule("@vaadin/vaadin-messages/src/vaadin-message-list.js")
+@NpmPackage(value = "@vaadin/vaadin-messages", version = "1.0.0-alpha1")
 public class MessageList extends Component implements HasStyle, HasSize {
 
+    private List<MessageListItem> items = Collections.emptyList();
+    private boolean pendingUpdate = false;
+
+    /**
+     * Creates a new message list component. To populate the content of the
+     * list, use {@link #setItems(Collection)}.
+     */
     public MessageList() {
-        getElement().appendChild(new Text("This is MessageList").getElement());
+    }
+
+    /**
+     * Creates a new message list component, with the provided items rendered as
+     * messages.
+     *
+     * @param items
+     *            the items to render as messages
+     * @see #setItems(Collection)
+     */
+    public MessageList(Collection<MessageListItem> items) {
+        setItems(items);
+    }
+
+    /**
+     * Creates a new message list component, with the provided items rendered as
+     * messages.
+     *
+     * @param items
+     *            the items to render as messages
+     * @see #setItems(MessageListItem...)
+     */
+    public MessageList(MessageListItem... items) {
+        setItems(items);
+    }
+
+    /**
+     * Sets the items that will be rendered as messages in this message list.
+     *
+     * @param items
+     *            the items to set
+     */
+    public void setItems(Collection<MessageListItem> items) {
```

Comment:
Should non-null items parameter be asserted? new ArrayList<>(null); will throw an exception anyhow, but could be good to be more specific about it.

Reply:
Good point. Done.